docs/bytecode.md

```diff
@@ -1,144 +1,32 @@
 # StableHLO Bytecode
 
-## Currently Encoded Attributes / Types
+## MLIR Bytecode Format
 
-### StableHLO Attributes and Types
+StableHLO uses the [MLIR Bytecode Format](https://mlir.llvm.org/docs/BytecodeFormat/)
+for artifact serialization.
 
-Documentation on the structure of the encoded attributes and types can be found
-in the following code comments:
+The MLIR Bytecode Format is a serialization format used to encode MLIR
+programs. It was built with high consideration for serialization
+and deserialization performance, disk and memory footprint, mmap-ability.
+Performance, serialization size, and memory tests were run using test
+files from upstream dialects and larger CIRCT test files to validate the
+format. MLIR bytecode was not specifically built to make MLIR stable, but
+the [MLIR RFC](https://discourse.llvm.org/t/rfc-a-binary-serialization-format-for-mlir/63518)
+notes that it would not be difficult to build stability on top of this format,
+which we successfully did for StableHLO in the [StableHLO Compatibility RFC](https://github.com/openxla/stablehlo/blob/main/rfcs/20220912-compatibility.md).
```

Comment:
Consider linking to compatibility.md instead, since it's designed to be easier to digest (and it also links to the RFC if the reader is interested in details).